source/extensions/filters/http/jwt_authn/jwks_cache.cc

```diff
@@ -0,0 +1,169 @@
+#include "extensions/filters/http/jwt_authn/jwks_cache.h"
+
+#include <chrono>
+#include <unordered_map>
+
+#include "common/common/logger.h"
+#include "common/config/datasource.h"
+
+using ::google::jwt_verify::Jwks;
+using ::google::jwt_verify::Status;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace JwtAuthn {
+namespace {
+
+// Default cache expiration time in 5 minutes.
+const int kPubkeyCacheExpirationSec = 600;
+
+// HTTP Protocol scheme prefix in JWT aud claim.
+const std::string kHTTPSchemePrefix("http://");
+
+// HTTPS Protocol scheme prefix in JWT aud claim.
+const std::string kHTTPSSchemePrefix("https://");
+
+/**
+ * Searches protocol scheme prefix and trailing slash from an audience string.
+ * returns one without these prefix and suffix for consistent comparison.
+ * @param input audience string.
+ * @return sanitized audience string without scheme prefix and tailing slash.
+ */
+std::string sanitizeAudience(const std::string& aud) {
+  int beg = 0;
```

Comment:
Nit: please use explicit sized types, e.g. `uint32_t` as per Envoy style, or if you're being compatible with an existing API, something like `size_t` as determined by the STL API.